src/main/java/com/gluonhq/gradle/tasks/ConfigBuild.java

```diff
@@ -97,7 +101,19 @@ public void build() {
     }
 
     private ProjectConfiguration createSubstrateConfiguration() {
-        ProjectConfiguration clientConfig = new ProjectConfiguration((String) project.getProperties().get("mainClassName"), getClassPath());
+        // Use Application Plugin First to get mainClass
+        Property<String> mainClass = project.getObjects().property(String.class);
+        project.getPlugins().withType(ApplicationPlugin.class, applicationPlugin -> {
+            JavaApplication javaApp = project.getExtensions().getByType(JavaApplication.class);
+            mainClass.set(javaApp.getMainClass().getOrNull());
+        });
+        // Fallback to deprecated mainClassName
+        if (!mainClass.isPresent()){
+            mainClass.set((String) project.getProperties().get("mainClassName"));
+        }
+
+        // Init Client Config
+        ProjectConfiguration clientConfig = new ProjectConfiguration(mainClass.getOrNull(), getClassPath());
```

Comment:
if neither `mainClass` nor `mainClassName` are found, a null is passed to `ProjectConfiguration` that has a check on it, and returns a NPE with message `Main class name is required` (which is fine, of course).
Another way:
```
       if (!mainClass.isPresent()) {
            throw new RuntimeException("Error: neither mainClass nor mainClassName are set");
        }

        // Init Client Config
        ProjectConfiguration clientConfig = new ProjectConfiguration(mainClass.get(), getClassPath());
...
```